src/main/cpp/blaze_util.cc

```diff
@@ -74,6 +76,26 @@ bool GetNullaryOption(const char *arg, const char *key) {
   return true;
 }
 
+std::vector<std::string> SearchNaryOption(const vector<string>& args,
+                              const char *key) {
+  vector<std::string> values;
+  for (vector<string>::size_type i = 0; i < args.size(); ++i) {
```

Comment:
I think you need to advance `i` extra if the flag value came from a `--bazelrc foo`. Adding a test for `--bazelrc --bazelrc foo` would help confirm/deny.

Reply:
Do you think this should be covered by this?
```
TEST_F(BlazeUtilTest, TestSearchNaryStartupOptionWithoutEquals2) {
  assert_equal_vector_char_pointer({"value1", "value2"}, SearchNaryOption({"bazel", "--flag", "value1", "--flag", "value2", "build", ":target"}, "--flag"));
}
```

or are you specifically mean `--bazelrc --bazelrc foo`? The current behavior is
```
$ bazel-bin/src/bazel --bazelrc --bazelrc 1.rc info
[ERROR 15:32:34.993 src/main/cpp/option_processor.cc:391] Error: Unable to read .bazelrc file '/Users/yic/workspace/bazel/--bazelrc'.
[FATAL 15:32:34.994 src/main/cpp/blaze.cc:1295]
```
I think this should be expected since every `--bazelrc` should expects a value?

Reply:
The latter case (`--bazelrc --bazelrc foo`). I think this currently returns `[--bazelrc, foo]`, when it should really return `[--bazelrc]`.

Reply:
With this change, the error 
```
$ bazel --bazelrc --bazelrc foo info
[ERROR 09:27:17.974 src/main/cpp/option_processor.cc:382] Error: Unable to read .bazelrc file '/Users/yic/workspace/bazel/--bazelrc'.
[FATAL 09:27:17.975 src/main/cpp/blaze.cc:1293] 
$ bazel --bazelrc --bazelrc foo 
[ERROR 09:27:20.867 src/main/cpp/option_processor.cc:382] Error: Unable to read .bazelrc file '/Users/yic/workspace/bazel/--bazelrc'.
[FATAL 09:27:20.867 src/main/cpp/blaze.cc:1293]
```
means it thinks `--bazelrc` is the value for key `--bazelrc`, and `foo` isn't in the picture, which is consistent with the behavior without the change:
```
$ bazel --bazelrc --bazelrc 1.rc info
[ERROR 09:26:19.832 src/main/cpp/option_processor.cc:382] Error: Unable to read .bazelrc file '/Users/yic/workspace/bazel/--bazelrc'.
[FATAL 09:26:19.833 src/main/cpp/blaze.cc:1293]
```

I will add a test case for this if I understand correctly here.

Reply:
oh ok. I see. Will adjust.

Reply:
Tests added. Thanks! That's a good catch!